Add preprocessing script with new config and updated README

mpetkevich opened the pull request.

mpetkevich committed:
Add preprocessing script with new config and updated README

VGSML commented on src/readmissions/preprocess.py:
```diff
@@ -0,0 +1,876 @@
+    """)
+
+def load_raw_inpatient_data(conn: duckdb.DuckDBPyConnection, file_path: str, table_name: str):
+    conn.query(f"DROP TABLE IF EXISTS {table_name};")
```
не уверен что нам нужен параметр с именем таблицы, мы все делаем в памяти, и потом это никуда не идет

VGSML commented on src/readmissions/preprocess.py:
```diff
@@ -0,0 +1,876 @@
+        GROUP BY patient_id, encounter_id, period, labs_values.code;
+    """)
+
+def load_vitals(
```
я думаю что нужно проверять указан ли путь, и если нет то не грузить данные для этого файла

VGSML replied:
и передавать не просто параметрами а словарем

VGSML commented on src/readmissions/preprocess.py:
```diff
@@ -0,0 +1,876 @@
+                ) TO '{pathlib.Path(output_path) / 'data.parquet'}' (FORMAT PARQUET);
+    """)
+
+def save_target(conn: duckdb.DuckDBPyConnection, output_path: str):
```
внеси изменения о которых мы вчера говорили

VGSML commented on src/readmissions/preprocess.py:
```diff
@@ -0,0 +1,876 @@
+    conn.query(
+        f"COPY dynamic_data_vocab TO '{pathlib.Path(output_path) / 'dynamic_data_vocab.parquet'}' (FORMAT PARQUET);")
+
+def save_data(conn: duckdb.DuckDBPyConnection, output_path: str):
```
И тут тоже

VGSML reviewed (changes requested):
Все так, осталось внести правки о которых мы вчера говорили

mpetkevich committed:
Update SQL queries, enhance limit processing, and apply cosmetic code improvements

mpetkevich committed:
Add time aggregation and optional vitals files support

- Added time-based aggregation (e.g. every 6 hours)
- Added support for optional vitals files via dictionary
- Simplified SQL queries and removed redundant expressions
- Improved config validation and logging output
